geo-traits/src/point.rs

```diff
@@ -0,0 +1,185 @@
+use std::marker::PhantomData;
+
+use geo_types::{Coord, CoordNum, Point};
+
+use crate::Dimensions;
+
+/// A trait for accessing data from a generic Point.
+///
+/// Refer to [geo_types::Point] for information about semantics and validity.
+pub trait PointTrait {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// Access the n'th (0-based) element of the CoordinateTuple.
+    /// May panic if n >= DIMENSION.
+    /// See also [`nth()`](Self::nth).
+    fn nth_unchecked(&self, n: usize) -> Self::T;
+
+    /// Dimensions of the coordinate tuple
+    fn dim(&self) -> Dimensions;
+
+    /// Access the n'th (0-based) element of the CoordinateTuple.
+    /// Returns NaN if `n >= DIMENSION`.
+    /// See also [`nth()`](Self::nth_unchecked).
+    fn nth(&self, n: usize) -> Option<Self::T> {
+        if n < self.dim().size() {
+            Some(self.nth_unchecked(n))
+        } else {
+            None
+        }
+    }
+
+    /// x component of this point.
+    fn x(&self) -> Self::T {
+        self.nth(0).unwrap()
```

Comment:
I didn't notice this default impl before - is it worth having a default impl? 

It will be wrong for lon/lat ordered geometries (admittedly only very occasionally).

And for things like geo-types, it seems like it'll be suboptimal. 

Admittedly it will save a couple lines of code in the common case, but it's very straight forward to implement.

I'm OK with keeping it if you think it's worthwhile.

Reply:
It was recently added. I'm happy to revert this and make this non-default.